docs/static/transforming-data.asciidoc

```diff
@@ -0,0 +1,520 @@
+filter {
+  ruby {
+    code => "event.cancel if rand <= 0.90"
+  }
+}
+--------------------------------------------------------------------------------
+
+
+[[data-deserialization]]
+=== Deserializing Data
+
+The plugins described in this section are useful for deserializing data into
+Logstash events.
+
+Also see <<filter-plugins>> and <<codec-plugins>> for the full list of available
+data processing plugins.
+
+<<plugins-codecs-avro,avro codec>>::
+
+Reads serialized Avro records as Logstash events. This plugin deserializes
+individual Avro records. It is not for reading Avro files. Avro files have a
+unique format that must be handled upon input.
++
+The following config deserializes input from Kafka:
++
+[source,json]
+----------------------------------
+input {
+  kafka {
+    codec => {
+      avro => {
+        schema_uri => "/tmp/schema.avsc"
+      }
+    }
+  }
+}
+...
+----------------------------------
+
+
+<<plugins-filters-csv,csv filter>>::
+
+Parses comma-separated value data into individual fields. By default, the
+filter autogenerates field names (column1, column2, and so on), or you can specify
+a list of names. You can also change the column separator.
++
+The following config parses CSV data into the field names specified in the
+`columns` field:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  csv {
+    separator => ","
+    columns => [ "Transaction Number", "Date", "Description", "Amount Debit", "Amount Credit", "Balance" ]
+  }
+}
+--------------------------------------------------------------------------------
+
+<<plugins-codecs-fluent,fluent codec>>::
+        
+Reads the Fluentd `msgpack` schema.
++
+The following config decodes logs received from `fluent-logger-ruby`:
++
+[source,json]
+--------------------------------------------------------------------------------
+input {
+  tcp {
+    codec => fluent
+    port => 4000
+  }
+}
+--------------------------------------------------------------------------------
+
+
+//ALVIN: SHOULD WE ALSO COVER THE MSGPACK CODEC? I LOOKED AT THE DOCS FOR MSGPACK AND THERE'S NO USEFUL INFO IN THERE. 
+
+<<plugins-codecs-json,json codec>>::
+
+Decodes (via inputs) and encodes (via outputs) JSON formatted content, creating
+one event per element in a JSON array.
++
+The following config decodes the JSON formatted content in a file:
++
+[source,json]
+--------------------------------------------------------------------------------
+input {
+  file {
+    path => "/path/to/myfile.json"
+    codec =>"json"
+}
+--------------------------------------------------------------------------------
+
+
+//ALVIN: SHOULD WE INCLUDE JSON_LINES HERE, TOO, OR MAYBE MENTION IT?
+
+<<plugins-codecs-protobuf,protobuf codec>>::
+
+Reads protobuf encoded messages and converts them to Logstash events. Requires
+the protobuf definitions to be compiled as Ruby files.
```

Comment:
@jordansissel We cover that in the docs for the protobuf codec. Not sure it makes sense to go into that level of detail in this particular topic. Although..I gotta say...I sure like the idea of recommending something called "beefcake" in the doc.s :-P